dotnet/src/SemanticKernel.Abstractions/SkillDefinition/SKFunctionContextParameterAttribute.cs

```diff
@@ -41,6 +41,11 @@ public string Name
     /// </summary>
     public string DefaultValue { get; set; } = string.Empty;
 
+    /// <summary>
+    /// Whether the parameter is required.
+    /// </summary>
+    public bool IsRequired { get; set; } = true;
+
     /// <summary>
     /// Creates a parameter view, using information from an instance of this class.
     /// </summary>
```

Comment:
#1195 is proposing some changes in the area. Might use this to help speed up things over there.